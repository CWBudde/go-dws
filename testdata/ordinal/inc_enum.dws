// Test Inc() with enum values
// Task 9.24: Inc() built-in function for enums

type TColor = (Red, Green, Blue, Yellow);

var c: TColor := Red;
PrintLn('Initial color ordinal: ', Ord(c));
PrintLn('Initial color value: ', Integer(c));

Inc(c);
PrintLn('After first Inc(): ', Ord(c));

Inc(c);
PrintLn('After second Inc(): ', Ord(c));

Inc(c);
PrintLn('After third Inc(): ', Ord(c));

// Test with enum with explicit values
type TStatus = (Ok = 0, Warning = 5, Error = 10);

var s: TStatus := Ok;
PrintLn('Status initial: ', Ord(s));

Inc(s);
PrintLn('After Inc (Ok to Warning): ', Ord(s));

Inc(s);
PrintLn('After Inc (Warning to Error): ', Ord(s));
