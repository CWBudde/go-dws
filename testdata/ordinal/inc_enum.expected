Initial color ordinal:  0
Initial color value:  0
After first Inc():  1
After second Inc():  2
After third Inc():  3
Status initial:  0
After Inc (Ok to Warning):  5
After Inc (Warning to Error):  10
